src/main/java/com/iexec/common/chain/IexecHubAbstractService.java

```diff
@@ -149,13 +171,22 @@ public IexecHubAbstractService(Credentials credentials,
         }
         this.retryDelay = nbBlocksToWaitPerRetry * (int)this.blockTime.toMillis();
         this.maxRetries = maxRetries;
+        this.iexecSmartContractValidator = iexecSmartContractValidator;
 
         String hubAddress = getHubContract().getContractAddress();
         log.info("Abstract IexecHubService initialized (iexec proxy address) " +
                 "[hubAddress:{}]", hubAddress);
         setMaxNbOfPeriodsForConsensus();
     }
 
+    @PostConstruct
+    private void checkSmartContractConnection() {
+        iexecSmartContractValidator.checkSmartContractConnection(
```

Comment:
Careful, in one the constructors (the [deprecated](https://github.com/iExecBlockchainComputing/iexec-common/pull/281/files#diff-fe771d01d22a765a39092fff6798eb71adb079c50f9e0ab2c1088eeafaa39279R138) one) `iexecSmartContractValidator` is null.

Reply:
Right, that was one point depending on the `expectedFinalDeadlineRatio` injection. Now it has been removed, we can have a default `iexecHubSmartContractValidator` for all constructors! :)